app/controllers/posts_controller.rb

```diff
@@ -2,10 +2,35 @@ class PostsController < ApplicationController
   def index
     user_id = params[:user_id]
     @user = User.find(user_id)
-    @posts = Post.includes(:user).where(user: params[:user_id]).references(:user)
+    @posts = @user.posts
   end
 
   def show
     @post = Post.includes(:user, comments: [:user]).find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.new(post_params)
```

Comment:
- [ ] On this method you are creating the post independently but the post belongs to a user, You should create a post the way it should belong to the current user. Please fix this issue. You can consider using this code `@post = current_user.posts.new(post_params)`.